pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,295 @@
+	if secretName != "" {
+		secret, err := c.secretLister.Secrets("openshift-config").Get(secretName)
+		if err != nil && !errors.IsNotFound(err) {
+			return "", nil, err
+		} else if errors.IsNotFound(err) {
+			conditions = []metav1.Condition{
+				{
+					Type:               "Degraded",
+					Status:             metav1.ConditionTrue,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "InvalidCustomSecret",
+					Message:            fmt.Sprintf("The %s/%s secret does not exist", "openshift-config", secretName),
+				},
+				{
+					Type:               "Progressing",
+					Status:             metav1.ConditionFalse,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "InvalidCustomSecret",
+					Message:            fmt.Sprintf("The %s/%s secret does not exist", "openshift-config", secretName),
+				},
+			}
+			secretName = ""
+		} else if errors := datasync.ValidateTLSSecret(secret); errors != nil {
+			conditions = []metav1.Condition{
+				{
+					Type:               "Degraded",
+					Status:             metav1.ConditionTrue,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "InvalidCustomSecret",
+					Message:            fmt.Sprintf("The %s/%s secret is invalid: %v", "openshift-config", secretName, errors),
+				},
+				{
+					Type:               "Progressing",
+					Status:             metav1.ConditionFalse,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "InvalidCustomSecret",
+					Message:            fmt.Sprintf("The %s/%s secret is invalid: %v", "openshift-config", secretName, errors),
+				},
+			}
+			secretName = ""
+		}
+	}
+	return secretName, conditions, nil
+}
+
+func (c *customRouteController) updateIngressConfigStatus(ctx context.Context, currentHostname string, conditions []metav1.Condition, ingress *configv1.Ingress) error {
+	desiredComponentRoute := configv1.ComponentRouteStatus{
+		Namespace:        c.componentRoute.Namespace,
+		Name:             c.componentRoute.Name,
+		DefaultHostname:  configv1.Hostname("oauth-openshift." + ingress.Spec.Domain),
+		CurrentHostnames: []configv1.Hostname{configv1.Hostname(currentHostname)},
+		ConsumingUsers: []configv1.ConsumingUser{
+			"system:serviceaccount:oauth-openshift:authentication-operator",
+		},
+		RelatedObjects: []configv1.ObjectReference{
+			{
+				Group:     routev1.GroupName,
+				Resource:  "routes",
+				Name:      "oauth-openshift",
+				Namespace: "openshift-authentication"},
+		},
+		Conditions: conditions,
+	}
+
+	if existingComponentRoute := common.GetComponentRouteStatus(ingress, c.componentRoute.Namespace, c.componentRoute.Name); existingComponentRoute == nil {
+		ingress.Status.ComponentRoutes = append(ingress.Status.ComponentRoutes, desiredComponentRoute)
+	} else {
+		existingComponentRoute = &desiredComponentRoute
+	}
+
+	_, err := c.ingressClient.UpdateStatus(ctx, ingress, metav1.UpdateOptions{})
```

Comment:
I plan to add a "deep equals" check to see if the conditions is the same (minus timestamps).

Reply:
you can have a look at https://github.com/openshift/library-go/blob/7b0fb0ed03f87f3b8db4e71b8ad9c0bf1ec5d3eb/pkg/operator/v1helpers/helpers.go#L157-L185 for inspiration, but yeah, you don't want to do "Updates" if the objects are the same

Reply:
Thanks and done!